controllers/druid/suite_test.go

```diff
@@ -1,196 +1,139 @@
+/*
+Copyright 2023 The Kubernetes authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// +kubebuilder:docs-gen:collapse=Apache License
+
 package druid
 
 import (
 	"context"
-	"errors"
-	"fmt"
 	"path/filepath"
 	"testing"
-	"time"
 
-	"github.com/stretchr/testify/require"
-	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
 
 	druidv1alpha1 "github.com/datainfrahq/druid-operator/apis/druid/v1alpha1"
-	// +kubebuilder:scaffold:imports
+	//+kubebuilder:scaffold:imports
 )
 
-type TestK8sEnvCtx struct {
-	restConfig *rest.Config
-	k8sManager manager.Manager
-	env        *envtest.Environment
-}
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
 
-func setupK8Evn(t *testing.T, testK8sCtx *TestK8sEnvCtx) {
-	ctrl.SetLogger(zap.New())
-	testK8sCtx.env = &envtest.Environment{
-		CRDInstallOptions: envtest.CRDInstallOptions{
-			Paths: []string{filepath.Join("..", "..", "config", "crd", "bases")},
-		},
-		ErrorIfCRDPathMissing: true,
-	}
+// +kubebuilder:docs-gen:collapse=Imports
 
-	config, err := testK8sCtx.env.Start()
-	require.NoError(t, err)
+/*
+Now, let's go through the code generated.
+*/
```

Comment:
nit: just for standardisation in the code base, single comments are addressed as ```//```, in case of a large block IMO its ok to have ```/*```